sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -78,26 +87,46 @@
    */
   public static <T, IdT> WithRepresentativeValues<T, IdT> withRepresentativeValueFn(
       SerializableFunction<T, IdT> fn) {
-    return new WithRepresentativeValues<T, IdT>(fn, null);
+    return new WithRepresentativeValues<>(fn, null);
+  }
+
+  private static <T> void validateWindowFn(PCollection<T> in) {
+    WindowingStrategy strategy = in.getWindowingStrategy();
+    checkArgument(strategy.getWindowFn().isNonMerging()
```

Comment:
I think this should be `UnsupportedOperationException` when false. It isn't that the user has made an error, but that we can't do the fairly reasonable thing they ask.